### PR TITLE
Hello World App in Quakus

### PR DESCRIPTION
**Pull request by shatakshi06**
Hi @somishraredhat @rsriniva @mabulgu

Please have a look at this pull request.

### CODE REVIEW

**Comment by mabulgu** [comment]
> Hi @somishraredhat @rsriniva @mabulgu
> 
> Please have a look at this pull request.

I will take a look ASAP. Thanks!

**Comment by mabulgu** [comment]
@shatakshi06 are the steps in the README.md valid? I want to go through the sample app so I wanted to be sure if the steps are valid or not. If they are not, can you update them so I can try out this demo. Thanks!

**Comment by shatakshi06** [comment]
> @shatakshi06 are the steps in the README.md valid? I want to go through the sample app so I wanted to be sure if the steps are valid or not. If they are not, can you update them so I can try out this demo. Thanks!

@mabulgu  Let me update it and let you know once it is done.

File: helloworld-simple-project/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# helloworld-simple-project Project
```
Comment (rsriniva) [inline helloworld-simple-project/README.md:1]: Please change the name of the top level folder to hello-quarkus and update this README to actually build, test and deploy the function - both locally as well as in the cluster using kn func emit etc

File: helloworld-simple-project/src/main/java/org/acme/GreetingResource.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.acme;
```
Comment (rsriniva) [inline helloworld-simple-project/src/main/java/org/acme/GreetingResource.java:1]: This file does not look like a serverless function? Is this what the kn create command created by default?

**Review by rsriniva — CHANGES_REQUESTED** [review]
Good start. See comments in PR inline

File: helloworld-simple-project/src/main/java/org/acme/GreetingResource.java (hunk 2 above)
Comment (shatakshi06) [reply helloworld-simple-project/src/main/java/org/acme/GreetingResource.java:1]: So yes I have created the function using kn create and did the changes in the business logic.

File: helloworld-simple-project/README.md (hunk 1 above)
Comment (shatakshi06) [reply helloworld-simple-project/README.md:1]: @rsriniva I have updated the readme file as per the mandatory process required to successfully run the serverless quarkus functions.

**Comment by shatakshi06** [comment]
> @shatakshi06 are the steps in the README.md valid? I want to go through the sample app so I wanted to be sure if the steps are valid or not. If they are not, can you update them so I can try out this demo. Thanks!

Hi @mabulgu I have updated the readme file with the procedure of serverless functions. Please Check.

File: helloworld-simple-project/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+This project uses Quarkus, the Supersonic Subatomic Java Framework.
+
+If you want to learn more about Quarkus, please visit its website: https://quarkus.io/ .
+
+
+## Creating a Serverless Function
+
+For Quarkus,
+                  kn func create -l quarkus -t http function-name
```
Comment (mabulgu) [inline helloworld-simple-project/README.md:11]: I think you can use code blocks for your readme. Also you can change your readme markdown file to an adoc file. That might be a good adoc practice for you.

here is our ref guide for adoc: https://github.com/RedHatTraining/skeleton/blob/main/documentation/ref_guide.adoc

Github might parse some things in a different way but most of them are valid for github too.

File: helloworld-simple-project/src/main/java/org/acme/GreetingResource.java (hunk 2 above)
Comment (mabulgu) [reply helloworld-simple-project/src/main/java/org/acme/GreetingResource.java:1]: you might also want to change the package name to sth like `com.redhat.training.examples`

File: helloworld-simple-project/src/main/java/org/acme/GreetingResource.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
```
Comment (mabulgu) [inline helloworld-simple-project/src/main/java/org/acme/GreetingResource.java:13]: Last time io remember suggesting sth about passing a parameter here and make it say `Hello Shatakshi` or sth for example. Can you try this out? Try to do this test-driven (via TDD I mean), so you should start with a test. I see that there are already some test skeleton is created , you can use that test classes and start with adding some more functions.

**Review by mabulgu — APPROVED** [review]
LGTM. I am approving this. Pls checkout the parameter passing thing and keeping using code blocks in the md file (or a better choice: converting it to adoc and using adoc for the readme - optional but might be good for your adoc learning)

Feel free to merge the PR btw.

**Review by rsriniva — APPROVED** [review]

File: helloworld-simple-project/README.md (hunk 3 above)
Comment (shatakshi06) [reply helloworld-simple-project/README.md]: Sure will check on this

File: helloworld-simple-project/src/main/java/org/acme/GreetingResource.java (hunk 4 above)
Comment (shatakshi06) [reply helloworld-simple-project/src/main/java/org/acme/GreetingResource.java]: Yes I will writing a different application for that and will commit it here.

**Pull request merged**